vsintegration/src/FSharp.Editor/CodeLens/CodeLens.fs

```diff
@@ -229,27 +228,27 @@ type CodeLensGeneralTagger (buffer:ITextBuffer) as self =
             layoutChangedCts <- new CancellationTokenSource()
             asyncMaybe {
                 do! Async.SwitchToContext uiContext |> liftAsync
-                // Suspend 5 ms, insantly applying the layout to the adornment elements isn't needed 
+                // Suspend 5 ms, instantly applying the layout to the adornment elements isn't needed 
                 // and would consume too much performance
                 do! Async.Sleep(5) |> liftAsync
                 let! view = view
                 let! layer = codeLensLayer
                 // Now relayout the existing adornments
-                if nonVisibleLineNumbers.Count > 0 || newVisibleLineNumbers.Count > 0 then
+                if nonVisibleLineNumbers.Count > 0 || newVisibleLineNumbers.Count > 0 || self.RelayoutRequested.Count > 0 then
+                    if self.RelayoutRequested.Count > 0 then self.RelayoutRequested.Dequeue() |> ignore
                     let buffer = buffer.CurrentSnapshot
                     for lineNumber in visibleLineNumbers do
                         let line = 
                             let l = buffer.GetLineFromLineNumber(lineNumber)
                             view.GetTextViewLineContainingBufferPosition(l.Start)
                         let tags = line.GetAdornmentTags(self)
                         match tags |> Seq.tryHead with
```

Comment:
You do this a few times. I suggest abstracting it into a function - also you can do the whole thing without needing as much indentation:

```f#
line.GetAdornmentTags(self)
|> Seq.tryHead
|> Option.bind (function :? StackPanel seq as uis -> Some uis | _ -> None)
|> Seq.filter (isNull >> not)
|> Seq.iter (fun ui ->
    ui.Visibility <- Visibility.Visible
    layoutUIElementOnLine view line.Extent ui
)
```

Reply:
you're right.